File: update_blog_entries.py
#!/usr/bin/env python3

import re
import requests

BLOG_URL = "http://azimut.github.io/blog"
LINE_START = "<li><b>"
LINE_REGEX = r"<b>(?P<date>\d{2}/\d{2})</b><a href=\"(?P<filename>.*.html)\">(?P<title>.*)</a>"

def main():
    response = requests.get(BLOG_URL)
    trs = generate_trs(response.text)
    print(trs)

def generate_trs(content: str) -> str:
    result = ""
    for line in content.splitlines():
        if not line.startswith(LINE_START): continue
        match = re.search(LINE_REGEX, line)
        if match:
            result += "<tr>"
            result += f"<td>{match.group('date')}</td>"
            result += f"<td><a target=\"_blank\" href=\"{BLOG_URL}/{match.group('filename')}\">{match.group('title')}</a></td>"
            result += "</tr>"
    return result

if __name__ == '__main__':
    main()

File: update_blog_entries.clj
(require '[babashka.pods :as pods])
(pods/load-pod 'retrogradeorbit/bootleg "0.1.9")
(require
 '[pod.retrogradeorbit.bootleg.utils :as bootleg]
 '[pod.retrogradeorbit.hickory.select :as s])

(def BLOG_URL "https://azimut.github.io/blog/")

(defrecord Entry [date filename title])

(defn get-blog []
  (-> BLOG_URL babashka.http-client/get :body str/trim))
(defn make-table []
  (->> (bootleg/convert-to (get-blog) :hickory)
       (s/select (s/descendant (s/class "org-ul") (s/tag :li)))
       (map :content)
       (map parse-entry)
       table))

(defn parse-entry [[bold anchor]]
  (->Entry (-> bold :content first)
           (-> anchor :attrs :href)
           (-> anchor :content first)))

(defn table [entries]
  (hiccup.core/html
   [:table {:id "blogs" :align "center"}
    (for [entry entries]
      (row entry))]))

(defn row [{:keys [date filename title]}]
  [:tr
   [:td date]
   [:td [:a {:href (str BLOG_URL filename) :target "_blank"}
         title]]])
